CollateralAccount/CollateralAccount.md

```diff
@@ -0,0 +1,106 @@
+---
+CIP: ?
+Title: Collateral Account for HD Wallets
+Authors: Sebastien Guillemot <seba@dcspark.io>
+Comments-URI: https://forum.cardano.org/t/collateral-account-derivation/65879
+Status: Draft
+Type: Standards
+Created: 2021-06-29
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This document describes using a separate derivation path to ensure there is always sufficient collateral to execute UTXO smart contracts
+
+# Motivation
+
+Collateral input | As of Alonzo, transactions that call non-native smart contracts are required to put up collateral to cover the potential cost of smart contract execution failure. Inputs used as collateral has the following properties:
+
+- Cannot contain any tokens (only ADA)
+- Cannot be a script address
+- Must be a UTXO input
+- Must be at least some percentage of the fee in the tx (concrete percentage decided by a protocol parameter)
+- Can be the same UTXO entry as used in regular tx input
+- Is consumed entirely (no change) if the contract execution fails during phase 2 validation
```

Comment:
I would add "Is not consumed if phase 2 validation succeeds"

Reply:
> I would add "Is not consumed if phase 2 validation succeeds"

That or specify if the collateral is passed along back as output.

Reply:
Note that a transaction input can be listed both in the transaction `txins` list and the collateral input list.